src/mlagility/common/filesystem.py

```diff
@@ -10,79 +14,79 @@
 else:
     DEFAULT_CACHE_DIR = os.path.expanduser("~/.cache/mlagility")
 
+CACHE_MARKER = ".mlacache"
+BUILD_MARKER = ".mlabuild"
 
-def full_model_path(corpora_dir, model_name):
+
+class CacheError(exc.GroqFlowError):
     """
-    Returns the absolute path of a model within a corpora
+    Raise this exception when the cache is being accessed incorrectly
     """
-    # FIXME: seems redundant with most of get_available_models()
 
-    file_name = f"{model_name}.py"
-    corpora = [
-        f
-        for f in os.listdir(corpora_dir)
-        if os.path.isdir(os.path.join(corpora_dir, f))
-    ]
-    for corpus in corpora:
-        full_path = f"{corpora_dir}/{corpus}/{file_name}"
-        if os.path.isfile(full_path):
-            return full_path
-    raise ValueError(f"Model {model_name} not found in tree {corpora_dir}")
 
+def make_build_dir(cache_dir: str, build_name: str):
+    # Create the build and cache directories, and put hidden files in them
+    # to mark them as such.
+    build_dir = build.output_dir(cache_dir, build_name)
+    os.makedirs(build_dir, exist_ok=True)
+
+    # File that indicates that the directory is an MLAgility cache directory
+    cache_file_path = os.path.join(cache_dir, CACHE_MARKER)
+
+    # File that indicates that the directory is an MLAgility build directory
+    build_file_path = os.path.join(build_dir, BUILD_MARKER)
+
+    with open(cache_file_path, mode="w", encoding="utf"):
+        pass
+
+    with open(build_file_path, mode="w", encoding="utf"):
+        pass
+
+
+def check_cache_dir(cache_dir: str):
+    cache_file_path = os.path.join(cache_dir, CACHE_MARKER)
+    if not os.path.isfile(cache_file_path):
+        raise CacheError(f"{cache_dir} is not a valid MLAgility cache")
```

Comment:
Can we be more specific here?
Something along the lines of `{cache_dir} has not been generated by MLAgility. You can only clean/delete directories that have been generated by MLAgility. Set a different --cache-dir before trying again.`

Alternatively, just lose the word "valid".